seacrowd/sea_datasets/parallel_asian_treebank/parallel_asian_treebank.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+from typing import List
+from itertools import combinations
+
+import datasets
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Tasks, Licenses
+
+_DATASETNAME = "parallel_asian_treebank"
+
+_LANGUAGES = ["khm", "lao", "mya", "ind", "fil", "zlm", "tha", "vie"]
+_LANGUAGES_TO_FILENAME_LANGUAGE_CODE = { "khm": "khm", "lao": "lo", "mya": "my", "ind": "id", "fil": "fil", "zlm": "ms", "tha": "th", "vie": "vi", }
+_LOCAL = True
+_CITATION = """\
+@inproceedings{riza2016introduction,
+  title={Introduction of the asian language treebank},
+  author={Riza, Hammam and Purwoadi, Michael and Uliniansyah, Teduh and Ti, Aw Ai and Aljunied, Sharifah Mahani and Mai, Luong Chi and Thang, Vu Tat and Thai, Nguyen Phuong and Chea, Vichet and Sam, Sethserey and others},
+  booktitle={2016 Conference of The Oriental Chapter of International Committee for Coordination and Standardization of Speech Databases and Assessment Techniques (O-COCOSDA)},
+  pages={1--6},
+  year={2016},
+  organization={IEEE}
+}
+"""
+
+_DESCRIPTION = """\
+The ALT project aims to advance the state-of-the-art Asian natural language processing (NLP) techniques through the open collaboration for developing and using ALT. 
+It was first conducted by NICT and UCSY as described in Ye Kyaw Thu, Win Pa Pa, Masao Utiyama, Andrew Finch and Eiichiro Sumita (2016). 
+Then, it was developed under ASEAN IVO. 
+The process of building ALT began with sampling about 20,000 sentences from English Wikinews, and then these sentences were translated into the other languages. 
+ALT now has 13 languages: Bengali, English, Filipino, Hindi, Bahasa Indonesia, Japanese, Khmer, Lao, Malay, Myanmar (Burmese), Thai, Vietnamese, Chinese (Simplified Chinese).
+"""
+
+_HOMEPAGE = "https://www2.nict.go.jp/astrec-att/member/mutiyama/ALT/"
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_URL = "https://www2.nict.go.jp/astrec-att/member/mutiyama/ALT/ALT-Parallel-Corpus-20191206.zip"
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ParallelAsianTreebank(datasets.GeneratorBasedBuilder):
+    """The ALT project aims to advance the state-of-the-art Asian natural language processing (NLP) techniques through the open collaboration for developing and using ALT"""
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=_SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_t2t",
```

Comment:
Is it intentional that this dataloader doesn't have a subset per language pair? @muhsatrio 

cc: @ljvmiranda921 @jamesjaya

Reply:
Adding english makes sense. My guess is that originally, the subsets were filtered only to Southeast asian languages. But maybe adding English as the "reference" language is helpful (if that's what you meant?)

Reply:
> Adding english makes sense. My guess is that originally, the subsets were filtered only to Southeast asian languages. But maybe adding English as the "reference" language is helpful (if that's what you meant?)

It's true that we're not intentionally collecting the english parts of the data, but it's just I don't know what language would work better as the "reference" in MT dataloaders. 🤔 Do you have any suggestion on this, @ljvmiranda921?

Reply:
For MT dataloaders in general, I don't know the answer. But for this dataset, I think using English makes sense because the way they collected the dataset is based on translations from English (The process of building ALT began with sampling about 20,000 sentences from English Wikinews, and then these sentences were translated into the other languages.). So yeah, in this dataset I think adding the pairs you mentioned makes a lot of sense. 👍

Reply:
Hi @muhsatrio @ljvmiranda921 @jamesjaya, apologies for the previous contradictory advice. 🙏 After discussing, this is what we agreed on.

For parallel MT dataloaders, we agreed upon having a subset for every possible direction with at least 1 SEA language.

For example, if the parallel MT dataloader has these languages: ind, sun, eng, spa, fil, mya, the dataloader consists of these subsets:
- `_{DATALOADERNAME}_eng_ind_seacrowd_t2t`
- `_{DATALOADERNAME}_eng_sun_seacrowd_t2t`
- `_{DATALOADERNAME}_eng_mya_seacrowd_t2t`
- `_{DATALOADERNAME}_ind_eng_seacrowd_t2t`
- `_{DATALOADERNAME}_sun_eng_seacrowd_t2t`
- `_{DATALOADERNAME}_mya_eng_seacrowd_t2t`
- `_{DATALOADERNAME}_spa_ind_seacrowd_t2t`
- `_{DATALOADERNAME}_spa_sun_seacrowd_t2t`
- `_{DATALOADERNAME}_spa_mya_seacrowd_t2t`
- `_{DATALOADERNAME}_ind_spa_seacrowd_t2t`
- `_{DATALOADERNAME}_sun_spa_seacrowd_t2t`
- `_{DATALOADERNAME}_mya_spa_seacrowd_t2t`
- `_{DATALOADERNAME}_ind_sun_seacrowd_t2t`
- `_{DATALOADERNAME}_ind_mya_seacrowd_t2t`
- `_{DATALOADERNAME}_sun_ind_seacrowd_t2t`
- `_{DATALOADERNAME}_mya_ind_seacrowd_t2t`
- `_{DATALOADERNAME}_sun_mya_seacrowd_t2t`
- `_{DATALOADERNAME}_mya_sun_seacrowd_t2t`

But since `eng` and `spa` are both non-SEA languages, no need to include `_{DATALOADERNAME}_eng_spa_seacrowd_t2t` and `_{DATALOADERNAME}_spa_eng_seacrowd_t2t`.

As similarly mentioned in #46.